kafka-client/src/test/java/dev/responsive/kafka/async/AsyncProcessorIntegrationTest.java

```diff
@@ -0,0 +1,293 @@
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Named;
+import org.apache.kafka.streams.processor.api.FixedKeyProcessor;
+import org.apache.kafka.streams.processor.api.FixedKeyProcessorContext;
+import org.apache.kafka.streams.processor.api.FixedKeyProcessorSupplier;
+import org.apache.kafka.streams.processor.api.FixedKeyRecord;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class AsyncProcessorIntegrationTest {
+
+  @RegisterExtension
+  static ResponsiveExtension EXTENSION = new ResponsiveExtension(StorageBackend.CASSANDRA);
+
+  private static final String INPUT_TOPIC = "input";
+  private static final String OUTPUT_TOPIC = "output";
+  private static final String ASYNC_KV_STORE = "async-kv";
+
+  private final Map<String, Object> responsiveProps = new HashMap<>();
+
+  private String name;
+  private Admin admin;
+
+  @BeforeEach
+  public void before(
+      final TestInfo info,
+      final Admin admin,
+      @ResponsiveConfigParam final Map<String, Object> responsiveProps
+  ) {
+    // add displayName to name to account for parameterized tests
+    name = info.getDisplayName().replace("()", "");
+
+    this.responsiveProps.putAll(responsiveProps);
+
+    this.admin = admin;
+    createTopicsAndWait(admin, Map.of(inputTopic(), 10, outputTopic(), 1));
+  }
+
+  @AfterEach
+  public void after() {
+    admin.deleteTopics(List.of(inputTopic(), outputTopic()));
+  }
+
+  private String inputTopic() {
+    return name + "." + INPUT_TOPIC;
+  }
+
+  private String outputTopic() {
+    return name + "." + OUTPUT_TOPIC;
+  }
+
+  @Test
+  public void shouldProcessEventsInOrderByKey() throws Exception {
+    // Given:
+    final List<String> keys = List.of("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k",
+                                      "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v",
+                                      "w", "x", "y", "z");
+
+    // produce 5 records for each key, with all keys intertwined between iterations
+    final List<KeyValue<String, String>> inputRecords = new LinkedList<>();
+    for (int val = 1; val < 6; ++val) {
+      for (final String key : keys) {
+        inputRecords.add(new KeyValue<>(key, key + val));
+      }
+    }
+
+    final Map<String, String> finalOutputRecords = new HashMap<>(keys.size());
+    for (final String key : keys) {
```

Comment:
It would be nice to make the number of iterations parameterizable so we can run a large number of iterations. This will make for a nice workload to run with antithesis.

Reply:
Yeah I agree, will do in a followup